public_profiler/v3.5-DD/ac.js

```diff
@@ -59,40 +59,29 @@ function ACCalc()
 }
 
 // void ACChangeArmor(void)
-// Copies the data from the armor table to the ac block and recalculates
-// the ac.
+// Copies the data from the armor table to the AC block and
+// recalculates the AC.
 function ACChangeArmor()
 {
   if (disable_autocalc())
     return;
 
   sheet().ACArmor.value  = 0;
-  if ( sheet().Armor1Worn.checked )
-  {
-      sheet().Armor1Carried.checked = true;
-      sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor1Bonus)));
-  }
-
-  if ( sheet().Armor3Worn.checked )
-  {
-      sheet().Armor3Carried.checked = true;
-      sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor3Bonus)));
-  }
-
-  if ( sheet().Armor4Worn.checked )
-  {
-      sheet().Armor4Carried.checked = true;
-          sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor4Bonus)));
-  }
-
   sheet().ACShield.value  = 0;
-  if ( sheet().Armor2Worn.checked )
-  {
-      sheet().Armor2Carried.checked = true;
-     sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
+
+  for (var i = 1; i <= 8; i++) {
```

Comment:
Might be a good idea to have a global constant for this, instead of just using 8 throughout.